Competition2017/src/competition/subsystems/climbing/RopeAlignerSubsystem.java

```diff
@@ -0,0 +1,39 @@
+package competition.subsystems.climbing;
+
+import org.apache.log4j.Logger;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import xbot.common.command.BaseSubsystem;
+import xbot.common.controls.actuators.XCANTalon;
+import xbot.common.injection.wpi_factories.WPIFactory;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+@Singleton
+public class RopeAlignerSubsystem extends BaseSubsystem {
+    private static Logger log = Logger.getLogger(RopeAlignerSubsystem.class);
+    
+    public final XCANTalon intakeMotor;
```

Comment:
Last year, I remember two cases where public members were used instead of the proper access methods (I believe one was with the arm motors and the other was some variety of limit switch); the year before, there was one case which spread throughout multiple subsystems. In those three cases, I was the one who fixed the logic to use the correct accessors. As I mentioned above, we haven't previously been catching these issues and this season there have been multiple problems that slipped past review as well. Will this particular instance be the one that is mis-used? Probably not. But there's a worryingly good chance that one of these cases where we're leaving things exposed out of simplicity will come back to bite us.

> Since your initial advocacy of this approach, when students ask how to start writing tests, I present them with two options - the public variable or the wrapper class, and talk a bit about the merits/disadvantages of both. Students naturally gravitate to the approach that makes more sense to them at the time - and thus, the one they're going to write/debug with more proficiency.

That's an error of method, not commentary on the idea. By the logic presented here, we should use an iterative robot next year and program it in Scratch.